bitcoin/src/taproot.rs

```diff
@@ -622,16 +630,14 @@ impl ScriptLeaf {
 }
```

Comment:
Shouldn't `ScriptLeaf::leaf_hash` return `TapNodeHash`? Do we even need `TapLeafHash` for anything now?

Reply:
We need `TapLeafHash` in taproot sign APIs. https://docs.rs/bitcoin/latest/bitcoin/util/sighash/struct.SighashCache.html#method.taproot_signature_hash

Reply:
Oh, right, in that case we should also have a trivial `impl From<TapLeafhash> for TapNodehash`.